jena-arq/src/main/java/org/apache/jena/query/ResultSetFactory.java

```diff
@@ -65,12 +66,12 @@ public static ResultSet load(String filenameOrURI) {
      * @param format
      * @return ResultSet
      */
-
     public static ResultSet load(String filenameOrURI, ResultsFormat format) {
         if (format == null)
             format = ResultsFormat.guessSyntax(filenameOrURI);
-        InputStream in = IO.openFile(filenameOrURI) ;
-        return load(in, format) ;
+        try ( InputStream in = IO.openFile(filenameOrURI) ) {
+            return load(in, format) ;
```

Comment:
I take it this doesn't prematurely close the `InputStream`, the runtime is smart enough to understand that there's still an active reference to the `InputStream` and not auto-close it in the happy path case?